File: .sem3/Podstawy programowania obiektowego/Include/towar.cpp
#include <string.h>

towar::towar()
{
	strcpy(nazwa," ");
	cena=0;
	sztuk=0;
}

void towar::czysc()
{
	strcpy(nazwa, " ");
	cena=0;
	sztuk=0;
}

void towar::setNazwa(char *naz)
{
	strcpy(nazwa,naz);
}
											//W WIEKSZOSCI KLAS TRZEBA MIEC WIECEJ SET'OW
bool towar::setCena(float cena)
{
	if(cena>0)
	{
		this->cena=cena; //this-> sluzy jako odnosnik do zmiennej, kt?ra nie b?dzie lokalna
		return 1;
	}
	else //nie musi
	{
		return 0;
	}
}

void towar::setSztuk(int szt) //powinno byc tak jak dla ceny
{
	sztuk=szt;	
}

void towar::pobierz(char *naz, float cen, int szt)
{
	strcpy(nazwa,naz);
	cena=cen;
	sztuk=szt;
}

char* towar::getNazwa()
{
	return nazwa;
}

float towar::getCena()
{
	return cena;
}

int towar::getSztuk()
{
	return sztuk;
}

float towar::wartosc()
{
	return cena*sztuk;
}

File: .sem4/Projektowanie Obiektowe/testproj/src/testproj/towar.java
package testproj;

public class towar 
{
	protected String nazwa;
	protected double cena;
	protected int sztuk;
	
	public towar()
	{
		nazwa="";
		sztuk=0;
		cena=0;
	}
	
	public towar(String naz, double cen, int szt)
	{
		nazwa=naz;
		sztuk=szt;
		cena=cen;
	}
	
	public void czysc()
	{
		nazwa="";
		sztuk=0;
		cena=0;
	}
	
	public void ustaw(String naz, double cen, int szt)
	{	//raczej NIE void
		nazwa=naz;
		sztuk=szt;
		cena=cen;
	}
	
	public void ustawNazwe(String naz)
	{	//NIE void
		nazwa=naz;
	}
	
	public void ustawSztuki(int szt)
	{	//NIE void
		sztuk=szt;
	}
	
	public void ustawCene(double cen)
	{	//NIE void
		cena=cen;
	}
	
	public String dajNazwe()
	{
		return nazwa;
	}
	
	public int dajSztuki()
	{
		return sztuk;
	}
	
	public double dajCene()
	{
		return cena;
	}
	
	public double wartosc()
	{
		return sztuk*cena;
	}
	
}
